src/diffusers/models/unet_2d_condition.py

```diff
@@ -595,18 +458,204 @@ def __init__(
             self.conv_norm_out = nn.GroupNorm(
                 num_channels=block_out_channels[0], num_groups=norm_num_groups, eps=norm_eps
             )
-
             self.conv_act = get_activation(act_fn)
+        else:
+            self.conv_norm_out = None
+            self.conv_act = None
 
+        conv_out_padding = (conv_out_kernel - 1) // 2
+        self.conv_out = nn.Conv2d(
+            block_out_channels[0], out_channels, kernel_size=conv_out_kernel, padding=conv_out_padding
+        )
+
+        self._set_pos_net_if_use_gligen(cross_attention_dim, attention_type)
```

Comment:
could we try to use keyword arguments here to be sure we don't accidentally pass the wrong value.

```suggestion
        self._set_pos_net_if_use_gligen(cross_attention_dim=cross_attention_dim, attention_type=attention_type)
```